### DISCUSSION TITLE
[Nuxt 3.12.4] Issue with Loading Private Environment Variables

### DISCUSSION THREAD

**Discussion by treighmawaka** [nuxt/nuxt#28643] [Questions] [answered]
Hi all, 

I recently upgraded my Nuxt application to version 3.12.4 and set it up for version 4 compatibility. I also reorganized the folder structure to align with the official recommendations. The app runs as expected, except for one problem: my private environment variables are no longer loading correctly. I didn’t experience this issue before the upgrade.

Here’s a snippet of how I’m trying to access the environment variables, but without success:
```ts
  // # nuxt.config.ts
  runtimeConfig: {
    // private env. variables
    SENDGRID_VERIFY_TEMPLATE_ID: process.env.SENDGRID_VERIFY_TEMPLATE_ID,
    
    // public env. variables
    public: {
      STRAPI_API_URL: `${process.env.STRAPI_URL}/api`,
    }
  }
})
```

```ts
const config = useRuntimeConfig();

const { STRAPI_API_URL, SENDGRID_VERIFY_TEMPLATE_ID } = config.public;

// /server/api/send.post.ts
export default defineEventHandler(async (event) => {	
	console.log({SENDGRID_VERIFY_TEMPLATE_ID, STRAPI_API_URL})
	
	// outputs endefined values👇 
	// {
	// 	SENDGRID_VERIFY_TEMPLATE_ID: undefined,
	// 	STRAPI_API_URL: undefined
	// }
});
```
Has anyone encountered something similar, or is there something specific I need to update in my configuration? Any advice would be appreciated!

**Comment by TheAlexLichter** [comment, marked as answer]
A few things:

1. Use `useRuntimeConfig` *inside* the `defineEventHandler`
2. You extract your private and public variable from `public` here
3. If you change variables at runtime, this setup will fail. See https://www.youtube.com/watch?v=_FYV5WfiWvs

**Reply by treighmawaka** [reply]
Oh wow!

I totally overlooked this. I moved `useRuntimeConfig` inside `defineEventHandler` and removed the private variables from the public property. 

Thanks so much!
